### PR TITLE
feat: Mill improvements

### PR DESCRIPTION
**Pull request by ckipp01**
This pr does a few different things to better support Mill. Each commit has a
full description, so it's best to view each of those to get the full idea.

### Test plan

There are new tests added to show that Java module support works, new tests to
show that you can now pass `--output dump.lsif` when using Mill, and the rest of
the tests should stay passing.

### CODE REVIEW

**Comment by jdorfman** [comment]
Thanks @ckipp01! We will get this reviewed ASAP.

File: semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/UriScheme.java [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 public enum UriScheme {
   DEFAULT,
+  /** @deprecated Use ZINC instead */
+  @Deprecated
```
Comment (olafurpg) [inline semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/UriScheme.java:6]: This is an internal API so no need for deprecation

**Review by olafurpg — APPROVED** [review]
LGTM 👍🏻 Thank you! 🙏🏻

**Comment by olafurpg** [comment]
https://github.com/sourcegraph/scip-java/releases/tag/v0.8.7

**Pull request merged**